### PR TITLE
Tabs: add component

### PR DESCRIPTION
**Pull request by zprong**
Figma:
https://www.figma.com/design/G3UM2urgAYO2iiNU7nlulI/%F0%9F%9F%A1-Cambio-Syntax?node-id=4540-1486&t=GyzLoUF9f8Sq4Pli-0

Pending:
- Waiting for [this ticket](https://cambly.atlassian.net/browse/SYN-62?atlOrigin=eyJpIjoiZDZkYmUyZjUyNDI0NGEyZjgzMTZmMjJmMGY2MDA4MjUiLCJwIjoiaiJ9) to be done to allow for `white-70` (text color on dark background) and `white-40` on the container's bottom border

Open questions [ANSWERED [BELOW](https://github.com/Cambly/syntax/pull/427#issuecomment-2123451151)]
- Typography is supposed to be 000-semibold-caps, but that's not supported (going to check in with AJ tomorrow)
- We don't have a white-secondary font, but Figma specifies it for unselected tabs on dark background
- Should dark background have the spanned underline across the tabs container?

Things to ignore:
- Design specifies a full-width bottom-border on the Tab bar if it changes URLs - AJ said we can skip this requirement for now if it adds too much complexity (but maybe it's not that hard?)

Tab Buttons
image

Tab Links
image

### CODE REVIEW

**Comment by zprong** [comment]
Question: would this be a major or minor bump?

**Comment by ajmooo** [comment]
1. yes, there’s a new secondary white color that has some transparency. I will file the bug in Jira now so that can get in. I’ll assign to @somethiiing so he’s aware
2. yes, the spanned underline should be in both on=lightBackground and on=darkBackground!

File: packages/syntax-core/src/Tabs/TabButton.module.css [hunk 1]
```diff
@@ -0,0 +1,18 @@
+.tab {
```
Comment (somethiiing) [inline packages/syntax-core/src/Tabs/TabButton.module.css:1]: id probably combine your 3 css files to just one, that way you can just reuse some of the styles. alot of them overlap for the most part (at least the classes do)

File: packages/syntax-core/src/Tabs/TabButton.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import { type ReactElement, type ComponentProps } from "react";
+import classnames from "classnames";
+import TapArea from "../TapArea/TapArea";
+import TabInternal from "./TabInternal";
+import styles from "./TabButton.module.css";
+
+export default function TabButton({
```
Comment (somethiiing) [inline packages/syntax-core/src/Tabs/TabButton.tsx:7]: i _feel_ like this should have an accessibility label as well?

File: packages/syntax-core/src/Tabs/TabInternal.module.css [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  justify-content: center;
+  padding: 4px;
+}
+
+.unselectedItemCount {
+  background-color: var(--color-cambio-gray-700);
+}
+
+.unselectedItemCountOnDarkBackground {
+  /* TODO: replace with white-70 */
```
Comment (somethiiing) [inline packages/syntax-core/src/Tabs/TabInternal.module.css:14]: why cant you do this now?

File: packages/syntax-core/src/Tabs/TabInternal.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    on === "lightBackground"
+      ? styles.unselectedItemCount
+      : styles.unselectedItemCountOnDarkBackground;
+  const selectedItemCountStyle =
+    on === "lightBackground"
+      ? styles.selectedItemCount
+      : styles.selectedItemCountOnDarkBackground;
+  const itemCountStyles = classnames(styles.itemCount, {
+    [unselectedItemCountStyle]: !selected,
+    [selectedItemCountStyle]: selected,
```
Comment (somethiiing) [inline packages/syntax-core/src/Tabs/TabInternal.tsx:26]: not sure what you prefer, but you can probably just combine those into one big classnames if you want.

```
  const itemCountStyles = classnames(styles.itemCount, {
    [styles.unselectedItemCount]: !selected && on==="lightBackground",
    [styles.unselectedItemCount]: !selected && on!=="lightBackground",
        ... etc
  })
```

File: packages/syntax-core/src/Tabs/TabInternal.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  });
+  return (
+    <div className={itemCountStyles}>
```
Comment (somethiiing) [inline packages/syntax-core/src/Tabs/TabInternal.tsx:29]: a little inclined to replace all of these styles with just a syntax Box and have a getBackground or something util to tweak your stylings.

if not, then i think at least wrap the outside with a <Box/> so you dont have to do the manual this stuff

```
.itemCount {
  border-radius: 999px;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 4px;
}
```

File: packages/syntax-core/src/Tabs/TabLink.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+   *
+   */
+  rel?: HtmlHTMLAttributes<HTMLAnchorElement>["rel"];
+  /**
+   * An optional onClick event. This is used for certain wrapper's support (such as react-router-dom).
+   */
+  onClick?: React.MouseEventHandler<HTMLAnchorElement>;
+};
+
+const TabLink = forwardRef<HTMLAnchorElement, TabLinkProps>(
```
Comment (somethiiing) [inline packages/syntax-core/src/Tabs/TabLink.tsx:36]: a reminder for when you do the syntax bump for this pr to also create expressTabButton + NextTabButton + update the eslint rules to do no restricted imports for this component

**Review by somethiiing — APPROVED** [review]
looks great, thanks for doing this! some comments are nits, but would take a look before merging

**Pull request merged**